tests/unit/amo/components/TestSearchContextCard.js

```diff
@@ -111,162 +134,393 @@ describe(__filename, () => {
 
     expect(root.find('.SearchContextCard-header')).toIncludeText(
-      'No add-ons found',
+      '0 results found',
     );
   });
 
-  it('should render singular form when only one result is found with addonType ADDON_TYPE_THEME', () => {
+  it('should render singular form when only one result is found with addonType ADDON_TYPE_THEMES_FILTER', () => {
+    const query = 'test';
     const { store } = dispatchSearchResults({
       addons: { [fakeAddon.slug]: fakeAddon },
       filters: {
-        addonType: ADDON_TYPE_THEME,
-        query: 'test',
+        addonType: ADDON_TYPE_THEMES_FILTER,
+        query,
+      },
+    });
+
+    const root = render({ store });
+
+    expect(root.find('.SearchContextCard-header')).toIncludeText(
+      `1 theme found for "${query}"`,
+    );
+  });
+
+  it('should render plural form when multiple results are found with addonType ADDON_TYPE_THEMES_FILTER', () => {
+    const query = 'test';
+    const { store } = dispatchSearchResults({
+      filters: {
+        addonType: ADDON_TYPE_THEMES_FILTER,
+        query,
       },
     });
 
     const root = render({ store });
 
     expect(root.find('.SearchContextCard-header')).toIncludeText(
-      '1 theme found for "test"',
+      `2 themes found for "${query}"`,
+    );
+  });
+
+  it('should fetch categories if there is a category filter', () => {
+    const { store } = dispatchSearchResults({
+      filters: { category: 'causes' },
+    });
+
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+
+    const root = render({ store });
+
+    sinon.assert.calledWith(
+      dispatchSpy,
+      fetchCategories({
+        errorHandlerId: root.instance().props.errorHandler.id,
+      }),
     );
   });
 
-  it('should render plural form when multiple results are found with addonType ADDON_TYPE_THEME', () => {
+  it('should not fetch categories if there is no category filter', () => {
     const { store } = dispatchSearchResults({
+      filters: { query: 'test' },
+    });
+
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+
+    render({ store });
+
+    sinon.assert.notCalled(dispatchSpy);
+  });
+
+  it("should not fetch categories if there is a category filter but it's already loading", () => {
+    const { store } = dispatchClientMetadata();
+
+    _searchStart({
+      store,
       filters: {
-        addonType: ADDON_TYPE_THEME,
+        category: 'causes',
+      },
+    });
+
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+
+    render({ store });
+
+    sinon.assert.notCalled(dispatchSpy);
+  });
+
+  it('should render results with categoryName and query for addonType ADDON_TYPE_THEMES_FILTER when loading is false', () => {
+    const categoryName = 'Causes';
+
+    const { store } = dispatchSearchResults({
+      filters: {
+        addonType: ADDON_TYPE_THEMES_FILTER,
+        category: 'causes',
         query: 'test',
       },
     });
 
+    _fetchCategories({ store });
+    _loadCategories({ store });
```

Comment:
I mentioned this in the code for the component as well, but I find this confusing, this test says "when loading is false", and the categories are not loading because you have fetched and loaded them, but the `loading` that is being used in the component is from the `search` state, not from the `categories` state.

Reply:
maybe I should put "...when search is loaded" ?. Is that any clearer?

I need to call the categorie functions in order to set the `categoryName`. But I am also using `dispatchSearchResults` which starts and loads the search. 

Does that make sense?

Reply:
I updated the tests a bit...let me know if that's better now